sympy/solvers/solvers.py

```diff
@@ -1968,7 +2022,7 @@ def key(sym):
     # so the docs should be updated to reflect that or else
     # the following should be `bool(failed) or not linear`
     default_simplify = bool(failed)
-    if flags.get('simplify', default_simplify):
+    if flags['simplify'] == True or flags['simplify'] is None and default_simplify:
```

Comment:
@smichr what is the purpose of this?

This is the only place where `simplify=True` and `simplify=None` are distinguished. If this was just `if flags['simplify']` then the None case for `simplify` could be removed.